app/uk/gov/hmrc/ctcguaranteebalancerouter/connectors/EISConnector.scala

```diff
@@ -93,15 +119,12 @@ class EISConnectorImpl(
           .copy(authorization = None, otherHeaders = Seq.empty)
           .withExtraHeaders(requestHeaders: _*)
 
-        withMetricsTimerResponse(MetricsKeys.eisAccessCodeEndpoint) {
-          httpClientV2
-            .post(url"${eisInstanceConfig.accessCodeUrl}")
-            .withBody(request)
-            .setHeader(headerCarrier.headersForUrl(headerCarrierConfig)(eisInstanceConfig.accessCodeUrl): _*)
+        withMetricsTimerResponse(metricsKey) {
+          call(headerCarrier)
             .execute[Either[UpstreamErrorResponse, HttpResponse]]
-            .map[Either[ConnectorError, AccessCodeResponse]] {
+            .map[Either[ConnectorError, A]] {
               case Right(httpResponse) =>
-                httpResponse.json.validate[AccessCodeResponse] match {
+                httpResponse.json.validate[A] match {
                   case JsSuccess(value, _) => Right(value)
                   case JsError(_)          => Left(ConnectorError.FailedToDeserialise)
                 }
```

Comment:
At the moment, the stub will return a 400 for a Bad Request but 500 for anything else - would we ever expect 400/404 from EIS? In the case that we do get a 400, how come we return 404?

Reply:
I'm going to try to get EIS to return the 404, and that's the ideal code to be returning -- hence why I left it in.